### ISSUE TITLE
Storage: uninitialized constant Signet::RemoteServerError

### ISSUE THREAD

**Issue by kryali** [GoogleCloudPlatform/google-cloud-ruby#2353]
Is there a wonky dependency thing happening here? I added the "google-cloud-storage" gem and it's failing to find the Signet error dependency.

```
2.4.3 :003 > require "google/cloud/storage"
 => true
2.4.3 :004 > Google::Cloud::Storage.new.bucket("kiran_test_bkt")
NameError: uninitialized constant Signet::RemoteServerError
	from (irb):4
```

**Comment by blowmage** [comment]
Can you post the error’s backtrace?

**Comment by blowmage** [comment]
I am away from a computer for the next several hours. Here is what I think is happening. This is an issue with dependencies. The gem `signet` was released and added the error class from the message in this issue. Then, `google-api-client` 0.23.5 was released and tries to use it, but dependency constraints won’t allow the latest version of `signet` to be used, even though the code is looking for the error class. I think this can be fixed if you can specify `google-api-client` 0.23.4 in your Gemfile. 

/cc @dazuma

**Comment by kryali** [comment]
You're right, downgrading `google-api-client` to 0.23.4 resolves this issue.

I'm afraid I don't have more of a stacktrace since this is from an interactive console.
